modules/packages/MPI.chpl

```diff
@@ -208,42 +208,37 @@ module MPI {
   }
 
   pragma "no doc"
-  record _initMPI {
-    var doinit : bool = false;
-    var freeChplComm : bool = false;
-
-    pragma "no doc"
-    proc deinit() {
-      if freeChplComm {
-        if numLocales > 1 {
-          coforall loc in Locales do on loc {
+  var _doinit : bool = false;
+  var _freeChplComm : bool = false;
+  // Module level deinit
+  proc deinit() {
+    if _freeChplComm {
+      if numLocales > 1 {
```

Comment:
While we're here, differentiating `numLocales > 1` vs. `numLocales == 1` shouldn't be necessary here. The `coforall` should cover both cases.